### PR TITLE
Resource module for ios_l2_interfaces

### PR DESCRIPTION
**Pull request by justjais**
##### SUMMARY

Resource module for `ios_l2_interfaces`, with supported operations and respective integration tests.
(ref model: [ios_l2_interfaces model](https://github.com/ansible-network/resource_module_models/pull/9))

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

ios_l2_interfaces
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by justjais** [comment]
recheck

File: lib/ansible/modules/network/ios/ios_l2_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = """
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ios/ios_l2_interfaces.py:38]: Add notes section with tested against version, the connection type to be used for this module and link to platform guide. Refer https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L106

Applicable to other ios modules (even the already merged resource modules)

**Review by ganeshrn — APPROVED** [review]

**Comment by justjais** [comment]
recheck

**Comment by samccann** [comment]
The module rst file itself isn't being generated for some reason. When I do: 
MODULES=ios_l2_interfaces make webdocs

I see:

[WARNING]: Could not parse ios_l2_interfaces due to 'ansible.parsing.yaml.objects.AnsibleUnicode object' has noattribute 'get'


Not sure what that means yet but posting here for now.

File: lib/ansible/modules/network/ios/ios_l2_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+#  negotiation auto
+
+"""
+
+RETURN = """
+before:
+  description: The configuration prior to the model invocation
+  returned: always
+  type: list
+  sample: The configuration returned will alwys be in the same format of the paramters above.
```
Comment (acozine) [inline lib/ansible/modules/network/ios/ios_l2_interfaces.py]: typos here (and I see the same typos elsewhere):
```suggestion
  sample: The configuration returned will always be in the same format of the parameters above.
```

File: lib/ansible/modules/network/ios/ios_l2_interfaces.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+DOCUMENTATION = """
+---
+module: ios_l2_interfaces
+version_added: 2.9
+short_description: Manage Layer-2 interface on Cisco IOS devices.
+description: This module provides declarative management of Layer-2 interface on Cisco IOS devices.
+author: Sumit Jaiswal (@justjais)
+notes:
+- Tested against Cisco IOSv Version 15.2 on VIRL
```
Comment (acozine) [inline lib/ansible/modules/network/ios/ios_l2_interfaces.py]: The YAML indentation looks wrong here:
```
notes:
- Tested against Cisco IOSv Version 15.2 on VIRL
```
Shouldn't all list items be indented two spaces from the header like this?
```
notes:
  - note 1
  - note 2
```
Looks like this formatting is used throughout.

**Review by acozine — COMMENTED** [review]
Typos, formatting question.

File: lib/ansible/modules/network/ios/ios_l2_interfaces.py (hunk 2 above)
Comment (justjais) [reply lib/ansible/modules/network/ios/ios_l2_interfaces.py]: updated

File: lib/ansible/modules/network/ios/ios_l2_interfaces.py (hunk 3 above)
Comment (justjais) [reply lib/ansible/modules/network/ios/ios_l2_interfaces.py]: updated

**Comment by justjais** [comment]
@samccann thanks for pointing out the fix, `type` field for access-vlan was not indented properly which was resulting into respective doc failure.

**Pull request merged**